src/api/Interpret.h

```diff
@@ -64,6 +64,55 @@ class DefinedFunctions {
     }
 };
 
+class TermNames {
+public:
+    bool has(std::string const & name) const { return nameToTerm.find(name) != nameToTerm.end(); }
+    bool has(PTRef term) const { return termToNames.find(term) != termToNames.end(); }
+
+    void insert(std::string const & name, PTRef term, bool scoped = false) {
+        assert(not has(name));
+        nameToTerm[name] = term;
+        termToNames[term].push_back(name);
```

Comment:
```suggestion
        termToNames.insert({term, {name}});
```
is slightly more efficient, although it does not look very nice

Reply:
This is not equivalent. Here we need the first version, because the key `term` may already exist in the map `termToNames`.

Reply:
You are right!